src/Polly/Caching/AsyncCacheEngine.cs

```diff
@@ -39,7 +40,7 @@ internal static class AsyncCacheEngine
         if (cacheHit)
         {
             onCacheGet(context, cacheKey);
-            return valueFromCache;
+            return valueFromCache!;
```

Comment:
Should this method return `TResult?` so that this isn't needed?

Reply:
i dont think so. if we have a cacheHit then we know valueFromCache is not null. so we can be sure ImplementationAsync will return a not null

Reply:
I thought we changed things based on my earlier comment to allow you to set `null` into the cache, so in that case you'd get a cache hit on a null.